idseq_pipeline/commands/host_filtering_functions.py

```diff
@@ -670,18 +692,32 @@ def uploader_check_wait_all():
             assert t.completed and not t.exception
     else:
         # run STAR
+        total_counts_from_star = {} # getting total_reads and file truncation status from STAR
         logparams = return_merged_dict(
             DEFAULT_LOGPARAMS,
             {"title": "STAR",
              "version_file_s3": STAR_BOWTIE_VERSION_FILE_S3,
              "output_version_file": os.path.join(RESULT_DIR, VERSION_OUT)})
-        run_and_log_s3(logparams, target_outputs["run_star"], lazy_run, SAMPLE_S3_OUTPUT_PATH, run_star, fastq_files, uploader_start)
+        run_and_log_s3(logparams, target_outputs["run_star"], lazy_run, SAMPLE_S3_OUTPUT_PATH,
+                       run_star, fastq_files, uploader_start, total_counts_from_star)
+        if not total_counts_from_star.get('total_reads'):
+            # Total reads not set. Most likely it's lazy run. Damn it. would have to actually count the reads.
+            # TODO: Remove this one we also lazy load the stats.json file
+            total_counts_from_star['total_reads'] = count_reads(fastq_files[0], initial_file_type_for_log)
+
+        stats.data.append(total_counts_from_star)
         stats.count_reads("run_star",
                           before_filename=fastq_files[0],
                           before_filetype=initial_file_type_for_log,
                           after_filename=os.path.join(RESULT_DIR, STAR_OUT1),
                           after_filetype=initial_file_type_for_log)
 
+        if total_counts_from_star.get('truncated'):
```

Comment:
There is a small subtlety here.    If this is a lazy rerun of something that was truncated, `total_counts_from_star` here wouldn't have the `truncated` key.   However, presumably, the `INPUT_TRUNCATED_FILE` would have been uploaded during the previous (non-lazy) run of STAR.   So this is undetectable from the webapp's point of view, but still I'd say it's a tricky scenario.

Reply:
In case someone in future adds logic the branches based on whether input was truncated or not, you might want to make the check for that here, not in the run_star function, so it can handle the lazyness.

And guess what?   When we get rid of run_and_log_S3, the run_star function will make its own lazyness determination, and will be able to handle this truncation signalling protocol perfectly.